From 0b65963957639892e7192d2c4b4ba546f87d007d Mon Sep 17 00:00:00 2001
From: Portisch <hugo.portisch@yahoo.de>
Date: Wed, 8 Jan 2020 14:39:15 +0100
Subject: [PATCH] osd: Do not change virtual resolution on CPU lower than GXM. 
  The Mali-T450 blob will see this as error and change the resolution   back
 to default.

---
 drivers/amlogic/media/osd/osd_fb.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/amlogic/media/osd/osd_fb.c b/drivers/amlogic/media/osd/osd_fb.c
index 595dc1e65996..11058836c44e 100644
--- a/drivers/amlogic/media/osd/osd_fb.c
+++ b/drivers/amlogic/media/osd/osd_fb.c
@@ -632,6 +632,11 @@ static int osd_check_var(struct fb_var_screeninfo *var, struct fb_info *info)
 	struct osd_fb_dev_s *fbdev = (struct osd_fb_dev_s *)info->par;
 	const struct color_bit_define_s *color_format_pt;
 
+	if (osd_meson_dev.cpu_id < __MESON_CPU_MAJOR_ID_GXM) {
+		var->xres_virtual = info->var.xres_virtual;
+		var->yres_virtual = info->var.yres_virtual;
+	}
+
 	fix = &info->fix;
 	color_format_pt = _find_color_format(var);
 	if (color_format_pt == NULL || color_format_pt->color_index == 0)
-- 
2.17.1
